src/main/java/io/nats/client/PublishOptions.java

```diff
@@ -45,15 +45,17 @@ public class PublishOptions {
     private final long expectedLastSeq;
     private final long expectedLastSubSeq;
     private final String msgId;
+    private final int msgTtlSeconds;
 
-    private PublishOptions(String stream, Duration streamTimeout, String expectedStream, String expectedLastId, long expectedLastSeq, long expectedLastSubSeq, String msgId) {
-        this.stream = stream;
-        this.streamTimeout = streamTimeout;
-        this.expectedStream = expectedStream;
-        this.expectedLastId = expectedLastId;
-        this.expectedLastSeq = expectedLastSeq;
-        this.expectedLastSubSeq = expectedLastSubSeq;
-        this.msgId = msgId;
+    private PublishOptions(Builder b) {
+        this.stream = b.stream;
+        this.streamTimeout = b.streamTimeout;
+        this.expectedStream = b.expectedStream;
+        this.expectedLastId = b.expectedLastId;
+        this.expectedLastSeq = b.expectedLastSeq;
+        this.expectedLastSubSeq = b.expectedLastSubSeq;
+        this.msgId = b.msgId;
+        this.msgTtlSeconds = b.msgTtlSeconds;

```

Comment:
since private I can change. simplified constructor to take builder like most other places